apps/confessions-client/pages/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+import { Confessions } from "../components/Confessions";
+import { Login } from "../components/Login";
+import { PublishConfession } from "../components/PublishConfession";
+
+export default function Page() {
+  const [accessToken, setAccessToken] = useState<string | null>(null);
+  const [newConfession, setNewConfession] = useState<string | undefined>();
+
+  useEffect(() => {
+    if (accessToken) return;
+
+    const token = window.localStorage.getItem("access_token");
+    setAccessToken(token);
+  }, [accessToken]);
+
+  const handleAccessToken = (token: string | null) => {
+    setAccessToken(token)
+    if (!token) {
+      window.localStorage.removeItem("access_token");
+    } else {
+      window.localStorage.setItem("access_token", token!);
+    }
+  }
```

Comment:
I think we should extract this hook/state, along with the logic that determines whether to show the user a login or logout button into its own component called `LoginLogout`. To make it not flicker (the flickering i pointed out earlier), you could change

```ts
const [accessToken, setAccessToken] = useState<string | null>(null);
```

to 
```ts
const [accessToken, setAccessToken] = useState<string | null>(() => window.localStorage.getItem("access_token") ?? null);
```

since localstorage is a synchronous API in web (thankfully)